src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -615,10 +609,8 @@ private async Task ComputeAnalyzerDiagnosticsAsync(AnalysisScope? analysisScope,
 
             Debug.Assert(analysisScope != null);
 
-            var originalAnalyzers = analysisScope.Analyzers;
             Compilation compilation;
             AnalyzerDriver? driver = null;
-            bool driverAllocatedFromPool = false;
```

Comment:
since the finally is gone, you can just move the declarations of compilation/driver to line 623?